File: A2SV-Community-Education/Div-3-Group-3-2024/LeetCode/Easy/941. Valid Mountain Array/Python solution/941. Valid Mountain Array.py
class Solution:
    def validMountainArray(self, arr: List[int]) -> bool:
        if len(arr) < 3:
            return False
        count= 0
        for i in range(1, len(arr)):
            if arr[0] > arr[1] or arr[i - 1] == arr[i]:
                return False
            if arr[i - 1] > arr[i]:
                count += 1
                for j in range(i + 1, len(arr)):
                    if arr[j - 1] <= arr[j]:
                        return False
                break
                
        return True if count >= 1 else False

File: A2SV-Community-Education/Div-3-Group-3-2024/LeetCode/Easy/941. Valid Mountain Array/C++ solution/941. Valid Mountain Array.cpp
class Solution {
public:
    bool validMountainArray(vector<int>& arr) {
        if (arr.size() < 3)
            return false;
        int count = 0;
        for (int i = 1; i < arr.size(); i++)
        {
            if (arr[0] > arr[1] || arr[i - 1] == arr[i])
                return false;
            if (arr[i - 1] > arr[i])
            {
                count++;
                for (int j = i + 1; j < arr.size(); j++)
                {
                    if (arr[j - 1] <= arr[j])
                        return false;
                }
                break;
            }
        }

        return count > 0;
    }
};